### PR TITLE
fix: support non-string groupbys for pie chart

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
This adds support for non-string labels to the Pie Chart Viz and uses `NULL_STRING` if a pie label is `None`. Also removes decimals for float groupbys that satisfy `is_integer()`. This is needed, as all `int` columns become `float` if they contain `NULL`s (Pandas limitation).

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### TEST PLAN
CI + new tests + local testing

### ADDITIONAL INFORMATION

- [x] Has associated issue: #10490 
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by ktmud — APPROVED** [review]
LGTM! Than's for the quick and very comprehensive fix!

File: tests/viz_tests.py [hunk 1]
```diff
@@ -1330,3 +1331,43 @@ def test_get_aggfunc_non_numeric(self):
             viz.PivotTableViz.get_aggfunc("strcol", self.df, {"pandas_aggfunc": "min"})
             == "min"
         )
+
+
+class TestDistributionPieViz(SupersetTestCase):
+    base_df = pd.DataFrame(
+        data={
+            "intcol": [1, 2, 3, 4],
+            "floatcol": [0.1, 0.2, 0.3, 0.4],
```
Comment (ktmud) [inline tests/viz_tests.py:1340]: Would it help to insert a couple of Nones in these columns as well? And maybe add a boolean column, too?

Comment (villebro) [reply tests/viz_tests.py:1340]: Good catch, numeric `null`s were not properly handled (showed up as `nan`), adding tests.

**Review by ktmud — APPROVED** [review]
looking good!

**Pull request merged**